packages/runtime-common/helpers/ai.ts

```diff
@@ -368,3 +369,93 @@ export function generateCardPatchCallSpecification(
     };
   }
 }
+
+export function getPatchTool(attachedOpenCard: CardDef, patchSpec: any) {
+  return {
+    type: 'function',
+    function: {
+      name: 'patchCard',
+      description: `Propose a patch to an existing card to change its contents. Any attributes specified will be fully replaced, return the minimum required to make the change. If a relationship field value is removed, set the self property of the specific item to null. When editing a relationship array, display the full array in the patch code. Ensure the description explains what change you are making.`,
+      parameters: {
+        type: 'object',
+        properties: {
+          card_id: {
+            type: 'string',
+            const: attachedOpenCard.id, // Force the valid card_id to be the id of the card being patched
+          },
+          description: {
+            type: 'string',
+          },
+          ...patchSpec,
+        },
+        required: ['card_id', 'attributes', 'description'],
+      },
+    },
+  };
+}
+
+export function getSearchTool(attachedOpenCard: CardDef) {
+  return {
+    type: 'function',
+    function: {
+      name: 'searchCard',
+      description: `Propose a query to search for a card instance related to module it was from. Always prioritise search based upon the card that was last shared. Ensure that you find the correct "module" and "name" from the OUTERMOST "adoptsFrom" field from the card data that is shared`,
+      parameters: {
+        type: 'object',
+        properties: {
+          card_id: {
```

Comment:
We need the card id to know what to patch, but the card id isn't something the search tool needs to have passed in, we can avoid passing this in completely

Reply:
If the ai returns this 

`{module: '../author', name: 'Author'}`

how does the host resolve to without knowing that `../author` is contextually relative to the Author instance

`http://localhost:4201/drafts/author`

My point is. The ai bot doesn't have sufficient data to deduce the correct search query by looking at serialized card data, we need to pass in more. Also, not knowing about the realms etc.

Reply:
We could start with only explaining catalogentry, we know that type and that would mean we can use the search to support creating new card types.

The next common use case I'd expect is finding things to link to, or finding things with no type filter at all.

Reply:
Resolved by getting by putting only absolute urls from serialized cards